src/datasets/packaged_modules/generator/generator.py

```diff
@@ -22,11 +22,21 @@ def __post_init__(self):
 class Generator(datasets.GeneratorBasedBuilder):
     BUILDER_CONFIG_CLASS = GeneratorConfig
 
+    def __init__(
+        self,
+        split: Optional[datasets.NamedSplit] = None,
+        **kwargs,
+    ):
+        self.split = split if split is not None else datasets.Split.TRAIN
+        return super().__init__(
+            **kwargs,
+        )
+
```

Comment:
Do not add the `split` parameter as an attribute of `Generator` class, but as an attribute of `GeneratorConfig` instead.